src/lightning/fabric/strategies/fsdp.py

```diff
@@ -125,10 +125,14 @@ class FSDPStrategy(ParallelStrategy, _Sharded):
             - ``"SHARD_GRAD_OP"``: Shards gradients and optimizer states only. Model parameters get replicated.
             - ``"NO_SHARD"``: No sharding (identical to regular DDP).
             - ``"HYBRID_SHARD"``: Shards model parameters, gradients, and optimizer states within a single machine, but
-              replicates across machines.
+              replicates across machines. See also the `device_mesh` parameter below.
 
             Also accepts a :class:`torch.distributed.fsdp.ShardingStrategy` enum value.
 
+        device_mesh: A tuple `(sharding size, replication size)` that defines over how many devices to shard and
```

Comment:
@Liyang90 Want to double check that the tuple ordering is correct. Looking at https://github.com/pytorch/pytorch/blob/main/torch/distributed/fsdp/_init_utils.py#L162-L163, it looks like the first dimension is for the inter pg and the second dimension is for the intra pg. By this logic, should it be "A tuple `(replication_size, sharding_size)`..." (i.e, swap the ordering of the tuple elements?

Reply:
Right. I have it corrected in my code, but haven't pushed it.